rust/private/rustc.bzl

```diff
@@ -33,6 +33,7 @@ CrateInfo = provider(
         "srcs": "List[File]: All source Files that are part of the crate.",
         "deps": "List[Provider]: This crate's (rust or cc) dependencies' providers.",
         "output": "File: The output File that will be produced, depends on crate type.",
+        "wasm_output": "File: The output File that will be produced, depends on crate type. Built for the wasm32-unknown-unknown target if requested.",
```

Comment:
It strikes me as unusual that the WASM target needs special fields in the standard rust rules and providers.

I suspect this was not your first choice. Can you tell me a bit about the rationale for this arrangement? Is this because the existing toolchain makes too many assumptions (about output, or dependency file types, or something else)?

Reply:
Right, this wasn't the ideal solution. It all ties together with your next comment about where we set the target triple to the exec triple so I'll try to address it there about why this first draft has multiple outputs for each rule.